mods/default/functions.lua

```diff
@@ -757,3 +757,34 @@ function default.can_interact_with_node(player, pos)
 
 	return false
 end
+
+local non_player_action_log = minetest.settings:get_bool("enable_non_player_action_log") or false
+
+function default.log_action(player, pos, message)
+	local who = player and player:get_player_name() or "(something)"
+	if player and not player.is_fake_player and player:is_player() then
+		-- log action of real player
+		minetest.log("action",  who .. " " .. message .. " at " .. minetest.pos_to_string(pos))
+	elseif non_player_action_log ~= false then
```

Comment:
Why this convoluted check when it must be a boolean - `true` or `false` and can't be `nil` anyways? Simply `elseif non_player_action_log` should suffice.